core/primitives/src/config.rs

```diff
@@ -0,0 +1,626 @@
+            storage_write_evicted_byte: 0,
+            storage_read_base: 0,
+            storage_read_key_byte: 0,
+            storage_read_value_byte: 0,
+            storage_remove_base: 0,
+            storage_remove_key_byte: 0,
+            storage_remove_ret_value_byte: 0,
+            storage_has_key_base: 0,
+            storage_has_key_byte: 0,
+            storage_iter_create_prefix_base: 0,
+            storage_iter_create_prefix_byte: 0,
+            storage_iter_create_range_base: 0,
+            storage_iter_create_from_byte: 0,
+            storage_iter_create_to_byte: 0,
+            storage_iter_next_base: 0,
+            storage_iter_next_key_byte: 0,
+            storage_iter_next_value_byte: 0,
+            touching_trie_node: 0,
+            promise_and_base: 0,
+            promise_and_per_promise: 0,
+            promise_return: 0,
+            validator_stake_base: 0,
+            validator_total_stake_base: 0,
+        }
+    }
+}
+
+/// Strongly-typed representation of the fees for counting.
+#[derive(Copy, Clone, Hash, PartialEq, Eq, Debug, PartialOrd, Ord)]
+#[allow(non_camel_case_types)]
+pub enum ExtCosts {
+    base,
+    contract_compile_base,
+    contract_compile_bytes,
+    read_memory_base,
+    read_memory_byte,
+    write_memory_base,
+    write_memory_byte,
+    read_register_base,
+    read_register_byte,
+    write_register_base,
+    write_register_byte,
+    utf8_decoding_base,
+    utf8_decoding_byte,
+    utf16_decoding_base,
+    utf16_decoding_byte,
+    sha256_base,
+    sha256_byte,
+    keccak256_base,
+    keccak256_byte,
+    keccak512_base,
+    keccak512_byte,
+    log_base,
+    log_byte,
+    storage_write_base,
+    storage_write_key_byte,
+    storage_write_value_byte,
+    storage_write_evicted_byte,
+    storage_read_base,
+    storage_read_key_byte,
+    storage_read_value_byte,
+    storage_remove_base,
+    storage_remove_key_byte,
+    storage_remove_ret_value_byte,
+    storage_has_key_base,
+    storage_has_key_byte,
+    storage_iter_create_prefix_base,
+    storage_iter_create_prefix_byte,
+    storage_iter_create_range_base,
+    storage_iter_create_from_byte,
+    storage_iter_create_to_byte,
+    storage_iter_next_base,
+    storage_iter_next_key_byte,
+    storage_iter_next_value_byte,
+    touching_trie_node,
+    promise_and_base,
+    promise_and_per_promise,
+    promise_return,
+    validator_stake_base,
+    validator_total_stake_base,
+}
+
+// Type of an action, used in fees logic.
+#[derive(Copy, Clone, Hash, PartialEq, Eq, Debug, PartialOrd, Ord)]
+#[allow(non_camel_case_types)]
+pub enum ActionCosts {
+    create_account,
+    delete_account,
+    deploy_contract,
+    function_call,
+    transfer,
+    stake,
+    add_key,
+    delete_key,
+    value_return,
+    new_receipt,
+}
+
+impl fmt::Display for ActionCosts {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", ActionCosts::name_of(*self as usize))
```

Comment:
There is some crate or macro that can reduce this code, e.g. https://docs.rs/enum-display-derive/0.1.0/enum_display_derive/